01-getting-started/projects/final/app/src/main/java/com/raywenderlich/podplay/viewmodel/PodcastViewModel.kt

```diff
@@ -0,0 +1,163 @@
+ *   or information technology.  Permission for such use, copying, modification,
+ *   merger, publication, distribution, sublicensing, creation of derivative works,
+ *   or sale is expressly withheld.
+ *   
+ *   This project and source code may use libraries or frameworks that are
+ *   released under various Open-Source licenses. Use of those libraries and
+ *   frameworks are governed by their own individual licenses.
+ *   
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ */
+
+package com.raywenderlich.podplay.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.raywenderlich.podplay.db.PodPlayDatabase
+import com.raywenderlich.podplay.db.PodcastDao
+import com.raywenderlich.podplay.model.Episode
+import com.raywenderlich.podplay.model.Podcast
+import com.raywenderlich.podplay.repository.PodcastRepo
+import com.raywenderlich.podplay.util.DateUtils
+import com.raywenderlich.podplay.viewmodel.SearchViewModel.PodcastSummaryViewData
+import java.util.*
+
+class PodcastViewModel(application: Application) : AndroidViewModel(application) {
+
+  var podcastRepo: PodcastRepo? = null
+  private val _podcastLiveData = MutableLiveData<PodcastViewData?>()
+  private var livePodcastSummaryData: LiveData<List<PodcastSummaryViewData>>? = null
+  val podcastLiveData: LiveData<PodcastViewData?> = _podcastLiveData
+  var activeEpisodeViewData: EpisodeViewData? = null
+
+  val podcastDao: PodcastDao = PodPlayDatabase
+      .getInstance(application, viewModelScope)
+      .podcastDao()
+
+  private var activePodcast: Podcast? = null
+
+  suspend fun setActivePodcast(feedUrl: String): PodcastSummaryViewData? {
+    val repo = podcastRepo ?: return null
+    val podcast = repo.getPodcast(feedUrl)
+    return if (podcast == null) {
+      null
+    } else {
+      _podcastLiveData.value = podcastToPodcastView(podcast)
+      activePodcast = podcast
+      podcastToSummaryView(podcast)
+    }
+  }
+
+  suspend fun getPodcast(podcastSummaryViewData: PodcastSummaryViewData) {
+    podcastSummaryViewData.feedUrl?.let { url ->
+      var podcast: Podcast?
+      // TODO 1
+      podcast = podcastRepo?.getPodcast(url)
+
+      podcast?.let {
+        // TODO 2
+        it.feedTitle = podcastSummaryViewData.name ?: ""
+        it.imageUrl = podcastSummaryViewData.imageUrl ?: ""
+        _podcastLiveData.value = podcastToPodcastView(it)
+        activePodcast = it
+      } ?: run {
+        _podcastLiveData.value = null
```

Comment:
this will not emit so there is no point in setting this

Reply:
Will simply unnecessary logic.